server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -71,6 +70,8 @@
   private final Map<String, CobolDocumentModel> docs = new ConcurrentHashMap<>();
   private final Map<String, CompletableFuture<List<DocumentSymbol>>> outlineMap =
       new ConcurrentHashMap<>();
+  private final Map<String, Future<Void>> futureMap = new ConcurrentHashMap<>();
+  private final ExecutorService threadPool = Executors.newCachedThreadPool();
```

Comment:
Default impl has a max pool size of Integer.MAX_VALUE. Can that be a problem?
suggestion?

Reply:
It won't be problem, because by default in one sec idle-thread will be killed. 
BUT, we are introducing another thread-pool here, so in LSP we have 3 more. We need to discuss it internally.